src/Signal.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Signal
+ * Handles the various ways in which an action can be resolved. Currently,
+ * it handles values, promises, and error-first callbacks
+ */
+
+function Signal (action, params, next) {
+  let strategy = action.length >= 2 ? withCallback : withValue
```

Comment:
so `action` is always either a function or an array with action function and a callback?